### PR TITLE
chore: Replace Well componenet with Card component

### PR DESCRIPTION
**Pull request by nikolagigic**
### SUMMARY
Replace react-bootstrap Well component with customised AntD Card component
https://github.com/apache/incubator-superset/issues/10254

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
- Edit Dataset
Screenshot 2020-12-24 at 14 38 36
- SQLab. Query History
Screenshot 2020-12-24 at 14 39 08
- SQLab View Keys and Indexes
Screenshot 2020-12-24 at 14 39 19

After:
- Edit Dataset
Screenshot 2020-12-24 at 14 33 58
- SQLab. Query History
Screenshot 2020-12-24 at 14 33 22
- SQLab View Keys and Indexes
Screenshot 2020-12-24 at 14 33 41

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/common/components/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { FunctionComponent } from 'react';
+// eslint-disable-next-line no-restricted-imports
+import { Card as AntdCard } from 'antd';
+
+interface CardProps extends React.Component {
```
Comment (geido) [inline superset-frontend/src/common/components/Card.tsx:23]: You could import the CardProps directly from antd/lib/card and extend those instead

File: superset-frontend/src/common/components/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  style: React.CSSProperties;
+  bodyStyle: React.CSSProperties;
+}
+
+const Card: FunctionComponent<CardProps> = ({ style, bodyStyle, children }) => {
```
Comment (geido) [inline superset-frontend/src/common/components/Card.tsx:28]: I think we are trying to enhance the Antdesign components using Emotion wherever is possible. Please have a look at common/components/Badge or common/components/Collapse for some examples of that

**Review by geido — CHANGES_REQUESTED** [review]
Looks good but I think it is deviating from the approach that we recently took to enhance the Antdesign components using Emotion. Please have a look at my comments

**Comment by junlincc** [comment]
Thanks for the PR @nikolagigic! 
@kgabryje Hey Kamil, please meet Nikola from Flexiana. i wonder if you mind reviewing each other's PRs 😀

**Comment by michael-s-molina** [comment]
@nikolagigic Thanks for the PR. Can you please reference #10254 in the PR description?

**Comment by kgabryje** [comment]
> Thanks for the PR @nikolagigic!
> @kgabryje Hey Kamil, please meet Nikola from Flexiana. i wonder if you mind reviewing each other's PRs 😀

Not a problem 🙂 Hello @nikolagigic!

File: superset-frontend/src/datasource/DatasourceEditor.jsx [hunk 3]
```diff
@@ -261,7 +262,11 @@ StackedField.propTypes = {
 };
 
 function FormContainer({ children }) {
-  return <Well style={{ marginTop: 20 }}>{children}</Well>;
+  return (
+    <Card style={{ marginTop: 20 }} bodyStyle={{ padding: 15 }}>
```
Comment (kgabryje) [inline superset-frontend/src/datasource/DatasourceEditor.jsx:266]: Can you move the inline styles to Emotion?

Comment (nikolagigic) [reply superset-frontend/src/datasource/DatasourceEditor.jsx:266]: There 2 cases of **Card** component where one requires different styling than the other. Default **Card** styling is in the Emotion already

File: superset-frontend/src/common/components/Card.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * under the License.
+ */
+import React from 'react';
+import { styled } from '@superset-ui/core';
+// eslint-disable-next-line no-restricted-imports
+import { Card as AntdCard } from 'antd';
+import { CardProps } from 'antd/lib/card';
+
+const Card = styled(({ ...props }: CardProps) => <AntdCard {...props} />)`
+  background-color: rgba(247, 247, 247, 1);
```
Comment (rusackas) [inline superset-frontend/src/common/components/Card.tsx:26]:
```suggestion
  background-color: ${({ theme }) => theme.colors.grayscale.light4};
```

File: superset-frontend/src/common/components/Card.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  border-radius: 5px;
```
Comment (rusackas) [inline superset-frontend/src/common/components/Card.tsx:27]:
```suggestion
  border-radius: ${({ theme }) => theme.borderRadius}px;
```

File: superset-frontend/src/common/components/Card.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+  .ant-card-body {
+    padding: 5px;
```
Comment (rusackas) [inline superset-frontend/src/common/components/Card.tsx:30]:
```suggestion
    padding: ${({ theme }) => theme.gridUnit}px;
```
This comes out to 4px, but that should probably look right.

File: superset-frontend/src/datasource/DatasourceEditor.jsx (hunk 3 above)
Comment (rusackas) [reply superset-frontend/src/datasource/DatasourceEditor.jsx:266]: It looks like we might not really _need_ that 20px top margin. 

As for the bodyStyle's padding, I would love to use the Theme variables here too. There are a few options:

1) use `withTheme` to use the `gridUnit`s here directly (4 gridUnits for 16px would be close enough)
2a) remove the bodyStyle prop and use a **class name** (e.g. 'padded') and use that className in the component's Emotion stylesheet to add the padding (again, with `gridUnit`s)
2b) remove the bodyStyle prop and use a **prop** (e.g. 'padded') and use that prop in the component's Emotion stylesheet to apply the padding (again, with `gridUnit`s)

I'd prefer either 2a or 2b, personally.

**Comment by geido** [comment]
@nikolagigic Can we export the Card from `src/common/components/index.tsx `and maybe add a bit of storybook as well?

**Review by rusackas — APPROVED** [review]
@nikolagigic Sorry this one has been sitting here so long, but I think we'll be OK to merge it now. It looks like all comments were addressed, so I've marked them as resolved, and I'm approving. 

I also saw the E2E tests were failing for some reason. I've closed/reopened to kick CI, so fingers crossed that it works 🤞 Please give this a rebase, and assess any failing Cypress tests if needed, and it should be good to go.

@junlincc Probably ought to give it a quick pass for Product/QA signoff before we merge, or delegate that job to anyone else she deems fit.

**Pull request closed without merging**